core/src/main/java/org/apache/brooklyn/core/catalog/internal/CatalogUtils.java

```diff
@@ -170,6 +173,13 @@ public static void installLibraries(ManagementContext managementContext, @Nullab
     public static String getCatalogItemIdFromLoader(BrooklynClassLoadingContext loader) {
         if (loader instanceof OsgiBrooklynClassLoadingContext) {
             return ((OsgiBrooklynClassLoadingContext)loader).getCatalogItemId();
+        } else if (loader instanceof BrooklynClassLoadingContextSequential) {
+            final Iterator<BrooklynClassLoadingContext> iterator = ((BrooklynClassLoadingContextSequential) loader).getPrimaries().iterator();
+            if (iterator.hasNext()) {
+                BrooklynClassLoadingContext osgiLoader = iterator.next();
+                return getCatalogItemIdFromLoader(osgiLoader);
+            }
+            else return null;
```

Comment:
Comment from @neykov ([here](https://github.com/neykov/brooklyn-server/pull/3#discussion_r79803301)):

Need to revisit this logic in light of latest updates.